src/test/java/com/commercetools/sync/services/impl/CustomObjectServiceImplTest.java

```diff
@@ -0,0 +1,246 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.sphere.sdk.client.BadRequestException;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+import io.sphere.sdk.customobjects.commands.CustomObjectUpsertCommand;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+import static io.sphere.sdk.customobjects.CustomObjectUtils.getCustomObjectJavaTypeForValue;
+import static io.sphere.sdk.json.SphereJsonUtils.convertToJavaType;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class CustomObjectServiceImplTest {
+
+    private SphereClient client = mock(SphereClient.class);
+
+    private CustomObjectServiceImpl service;
+    private List<String> errorMessages;
+    private List<Throwable> errorExceptions;
+
+    private String customObjectId;
+    private String customObjectContainer;
+    private String customObjectKey;
+
+    @BeforeEach
+    void setup() {
+        customObjectId = RandomStringUtils.random(15, true, true);
+        customObjectContainer = RandomStringUtils.random(15, true, true);
+        customObjectKey = RandomStringUtils.random(15, true, true);
+
+        errorMessages = new ArrayList<>();
+        errorExceptions = new ArrayList<>();
+        CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder.of(client)
+            .errorCallback((exception, oldResource, newResource, updateActions) -> {
+                errorMessages.add(exception.getMessage());
+                errorExceptions.add(exception.getCause());
+            })
+            .build();
+
+        service = new CustomObjectServiceImpl(customObjectSyncOptions);
+    }
+
+    @AfterEach
+    void cleanup() {
+        reset(client);
+    }
+
+    private interface CustomObjectPagedQueryResult extends PagedQueryResult<CustomObject> {
+    }
+
+    @Test
+    void fetchCachedCustomObjectId_WithKeyAndContainer_ShouldFetchCustomObject() {
```

Comment:
Is this unit test testing the same ting as `fetchCachedCustomObjectId_WithExistingCustomObject_ShouldFetchCustomObjectAndCache` in IT? If so, I would remove one of the tests. Recently we removed integration tests that were already covered by unit tests, so it's not good to reintroduce again the same problem (duplicate unit and integration tests). Check also other tests in this class for possible duplicates.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/557/commits/de95f359c568dbd9c08336dce2470caafa342faa

Reply:
Can you also add a reason for @Disabled?

Reply:
I removed the whole method already